application/src/main/java/org/eclipse/hono/application/ApplicationConfig.java

```diff
@@ -36,20 +38,36 @@
 @Configuration
 public class ApplicationConfig {
 
+    private final VertxOptions vertxOptions;
+
     /**
-     * Gets the singleton Vert.x instance to be used by Hono.
-     * 
-     * @return the instance.
+     * Adds metrics options to the Vert.x configuration, to write the Vert.x metrics to the configured Hono metrics server
+     *
+     * @param metricsOptions
      */
-    @Bean
-    public Vertx vertx() {
-        VertxOptions options = new VertxOptions()
+    @Autowired(required = false)
+    public ApplicationConfig(final MetricsOptions metricsOptions) {
```

Comment:
Since vertx() is called directly from other methods in the class I do not think so!?